cropper/app/lib/CropStore.scala

```diff
@@ -56,15 +58,19 @@ class CropStore(config: CropperConfig) extends S3ImageStorage(config) {
           val updatedCrop = for {
             // Note: if any is missing, the entry won't be registered
             source <- userMetadata.get("source")
-            x      <- userMetadata.get("bounds_x").map(_.toInt)
-            y      <- userMetadata.get("bounds_y").map(_.toInt)
-            w      <- userMetadata.get("bounds_w").map(_.toInt)
-            h      <- userMetadata.get("bounds_h").map(_.toInt)
+
+            // we've moved to kebab-case as localstack doesn't like `_`
+            // fallback to reading old values for older crops
+            // see https://github.com/localstack/localstack/issues/459
+            x      <- Try(userMetadata.getOrElse("bounds-x", userMetadata("bounds_x"))).toOption.map(_.toInt)
```

Comment:
Wrap in a `Try` as the `getOrElse` fallback of `userMetadata("bounds_x")` isn't guaranteed to succeed.

Reply:
I'd like to see this wrapped in a function.

Reply:
I did this but it made it less readable and was less obvious what was going on imo:

```scala
private def getMetadataValue(map: Map[String, String], preferredKey: String, fallbackKey: String): Option[String] = {
  Try(map.getOrElse(preferredKey, map(fallbackKey))).toOption
}

val updatedCrop = for {
  x <- getMetadataValue(userMetadata, "bounds-x", "bounds_x").map(_.toInt)
  y <- getMetadataValue(userMetadata, "bounds-y", "bounds_y").map(_.toInt)
  ... etc
}
```

Reply:
I think this is down to naming; `getOrElseOrNone` makes it clearer.